app/controllers/places_controller.rb

```diff
@@ -221,11 +221,11 @@ def create
       assign_geometry_from_file
     elsif !params[:geojson].blank?
       @geometry = geometry_from_geojson( params[:geojson] )
-      @place.validate_with_geom( @geometry, user: current_user )
+      @place.validate_with_geom( @geometry, max_area_km2: max_area_km2, max_observation_count: max_observation_count )
     end
 
     if @geometry # && @place.valid?
-      @place.save_geom( @geometry, user: current_user )
+      @place.save_geom( @geometry, max_area_km2: max_area_km2, max_observation_count: max_observation_count )
```

Comment:
I think this is the source of your spec failures, b/c the second hash arg of `save_geom()` is assumed to be extra attributes to be saved with the `PlaceGeometry` record. I can get specs to pass by removing those attributes from the has before [as we're doing with `user`](https://github.com/inaturalist/inaturalist/blob/main/app/models/place.rb#L474), but maybe it would be better to explicitly separate the extra `PlaceGeometry` attributes from options used by `save_geom()`, e.g. `def save_geom( geom, other_attrs = {}, options = {} )`.

Reply:
Thanks. There was something funny with my test database preventing me from running many tests. Fixed now.

Let me know what you think of [f3bae47](https://github.com/inaturalist/inaturalist/pull/3785/commits/f3bae47d3d4b2f7ba922a258b33d3ca202714b85).

I think the root of the problem is that the responsibilities of data construction, normalization, validation, and fault-tolerant persistence are kind of muddled together in the Place model. That's pretty common for Rails code, and consistent with the rest of the code base, but I think refactoring this and related code would make it more concise and easier to understand.